src/main/java/org/sonarlint/intellij/core/SonarQubeEventNotifications.java

```diff
@@ -0,0 +1,140 @@
+import org.sonarsource.sonarlint.core.client.api.common.NotificationConfiguration;
+import org.sonarsource.sonarlint.core.client.api.connected.ServerConfiguration;
+import org.sonarsource.sonarlint.core.client.api.notifications.SonarQubeNotification;
+import org.sonarsource.sonarlint.core.client.api.notifications.SonarQubeNotificationListener;
+import org.sonarsource.sonarlint.core.client.api.notifications.SonarQubeNotificationTime;
+import org.sonarsource.sonarlint.core.notifications.SonarQubeNotifications;
+
+public class SonarQubeEventNotifications extends AbstractProjectComponent {
+  public static final String GROUP_SONARQUBE_EVENT = "SonarLint: SonarQube Events";
+
+  private final ProjectBindingManager bindingManager;
+  private final SonarLintProjectState projectState;
+  private final SonarLintProjectSettings projectSettings;
+  private final EventListener eventListener;
+  private final SonarQubeNotificationTime notificationTime;
+  private final MessageBusConnection busConnection;
+
+  public SonarQubeEventNotifications(Project project, ProjectBindingManager bindingManager, SonarLintProjectState projectState,
+    SonarLintProjectSettings projectSettings) {
+    super(project);
+    this.bindingManager = bindingManager;
+    this.projectState = projectState;
+    this.projectSettings = projectSettings;
+    this.eventListener = new EventListener();
+    this.notificationTime = new ProjectNotificationTime();
+    this.busConnection = project.getMessageBus().connect(myProject);
+  }
+
+  @Override
+  public void projectOpened() {
+    register(projectSettings);
+    busConnection.subscribe(ProjectConfigurationListener.TOPIC, settings -> {
+      // always reset notification date, whether bound or not
+      projectState.setLastEventPolling(ZonedDateTime.now());
+      register(settings);
+    });
+  }
+
+  private void register(SonarLintProjectSettings settings) {
+    SonarQubeNotifications.get().remove(eventListener);
+    SonarQubeServer server;
+    try {
+      server = bindingManager.getSonarQubeServer();
+    } catch (InvalidBindingException e) {
+      // do nothing
+      return;
+    }
+    if (settings.isBindingEnabled() && server.enableNotifications()) {
+      NotificationConfiguration config = createConfiguration(settings, server);
+      SonarQubeNotifications.get().register(config);
+    }
+  }
+
+  private NotificationConfiguration createConfiguration(SonarLintProjectSettings settings, SonarQubeServer server) {
+    String projectKey = settings.getProjectKey();
+    ServerConfiguration serverConfiguration = SonarLintUtils.getServerConfiguration(server);
+    return new NotificationConfiguration(eventListener, notificationTime, projectKey, serverConfiguration);
+  }
+
+  /**
+   * Read and save directly from the mutable object.
+   * Any changes in the project settings will affect the next request.
+   */
+  private class ProjectNotificationTime implements SonarQubeNotificationTime {
```

Comment:
I disagree. SonarLintProjectState represents a setting in IntelliJ and is seralizable - I think it should be kept clean. NotificationTime is a wrapper to comply with SonarLint core API.